index.js

```diff
@@ -1,19 +1,20 @@
 'use strict';
 module.exports = function (str, opts) {
-	if (typeof str !== 'string') {
-		throw new TypeError('Expected a string in the first argument, got ' + typeof str);
+	const typeOfInitialArg = typeof str;
+	if (typeOfInitialArg !== 'string') {
+		throw new TypeError('Expected a string in the first argument, got ' + typeOfInitialArg);
 	}
 
-	var preserveNewlines = opts && opts.preserveNewlines;
+	const lineMatcher = /\r?\n/;
```

Comment:
Don't really see any added value in extracting this. Line 14 is really ugly now. So just revert it.